### PR TITLE
Fedekunze/redelegation txs

### PR DESCRIPTION
**Pull request by fedekunze**
Added redelegation for Tx history

Design: https://app.zeplin.io/project/59e66beff6721ea8bf562e5b/screen/5b8d4970a0f39c423dc9c4dc

Implementation from this PR:

screen shot 2018-09-19 at 5 33 37 pm

### CODE REVIEW

File: src/components/TmLiAnyTransaction/TmLiAnyTransaction.vue [hunk 1]
```diff
@@ -9,7 +9,7 @@ tm-li-transaction(v-else :color="colors.grey" :time="transaction.time" :block="t
 import TmLiBankTransaction from "../TmLiBankTransaction/TmLiBankTransaction"
 import TmLiStakeTransaction from "../TmLiStakeTransaction/TmLiStakeTransaction"
 import TmLiTransaction from "../TmLiTransaction/TmLiTransaction"
-import colors from "../TmLiTransaction/tranaction-colors.js"
```
Comment (faboweb) [inline src/components/TmLiAnyTransaction/TmLiAnyTransaction.vue]: <.< ups

File: src/components/TmLiStakeTransaction/TmLiStakeTransaction.vue [hunk 2]
```diff
@@ -8,6 +8,20 @@ tm-li-transaction(:color="color" :time="transaction.time" :block="transaction.he
...
+        b {{transaction.redelegation.amount}}
+        span &nbsp;{{transaction.redelegation.denom.toUpperCase()}}S
+      template(v-else)
+        b {{tx.shares}}
+        span &nbsp;Shares
+    div(slot="details")
+      | From&nbsp;
+      router-link(:to="this.validatorURL + '/' + tx.validator_src_addr") {{moniker(tx.validator_src_addr)}}
+      |  to&nbsp;
+      router-link(:to="this.validatorURL + '/' + tx.validator_src_addr") {{moniker(tx.validator_dst_addr)}}
```
Comment (faboweb) [inline src/components/TmLiStakeTransaction/TmLiStakeTransaction.vue]: validator_src_addr needs to be validator_dst_addr here

File: src/views/Components.vue [hunk 3]
```diff
@@ -56,6 +56,7 @@
       <tm-li-stake-transaction :transaction="Object.assign(JSON.parse(JSON.stringify(txs[4])), { state: 'locked' })" :validators="validators" unbonding_time="2592000000" v-on:end-unbonding="log('end unbonding')"/>
       <tm-li-stake-transaction :transaction="txs[4]" :validators="validators" unbonding_time="2592000000" v-on:end-unbonding="log('end unbonding')"/>
       <tm-li-stake-transaction :transaction="Object.assign(JSON.parse(JSON.stringify(txs[4])), { state: 'ended' })" :validators="validators" unbonding_time="2592000000" v-on:end-unbonding="log('end unbonding')"/>
+      <tm-li-stake-transaction :transaction="txs[5]" :validators="validators" v-on:end-redelegation="log('end redelegation')"/>
```
Comment (faboweb) [inline src/views/Components.vue]: why do we need the end-redelegation event?

File: src/components/TmLiStakeTransaction/TmLiStakeTransaction.vue (hunk 2 above)
Comment (fedekunze) [reply src/components/TmLiStakeTransaction/TmLiStakeTransaction.vue]: good catch, thanks !

File: src/views/Components.vue (hunk 3 above)
Comment (fedekunze) [reply src/views/Components.vue]: Technically we need a BeginRedelegate and CompleteRedelegate:

https://github.com/cosmos/cosmos-sdk/blob/10b916eb28a8aed07df80d8c305a40b7f073d739/x/stake/types/msg.go#L204-L209

https://github.com/cosmos/cosmos-sdk/blob/10b916eb28a8aed07df80d8c305a40b7f073d739/x/stake/types/msg.go#L265-L270

Comment (fedekunze) [reply src/views/Components.vue]: I'll open an issue on voyager for this

**Pull request merged**